### PR TITLE
Small improvements for tools chapter

### PR DESCRIPTION
**Pull request by bernhardmgruber**

### CODE REVIEW

File: talk/tools/compiling.tex [hunk 1]
```diff
@@ -64,11 +64,11 @@
...
-      \item[-Werror] the way to force yourself to look at warnings
+      \item[-Wall -Wextra] get all warnings
+      \item[-Werror] force yourself to look at warnings
     \end{description}
   \end{block}
-  \begin{block}{Around optimization}
+  \begin{block}{Optimization}
     \begin{description}
       \item[-g] add debug symbols
       \item[-O0, -O2] 0 = no optimization, -O2 = optimized
```
Comment (bernhardmgruber) [inline talk/tools/compiling.tex:74]: I wonder why we do not list `-O3` here. Until very recently you would not get vectorization with `-O2` so it is an important flag for my use cases.

File: talk/tools/editors.tex [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
     \begin{description}
     \item[\href{http://www.microsoft.com/}{\beamergotobutton{Visual Studio}}]
-      the Microsoft way
+      heavy, fully fledged IDE for Windows
     \item[\href{https://code.visualstudio.com/}{\beamergotobutton{Visual Studio Code}}]
-      more lightweight, open source and portable
+      lightweight editor, open source, portable, many plugins
     \item[\href{https://www.eclipse.org/}{\beamergotobutton{Eclipse}}]
-      similar, also open source and portable
+      IDE, open source, portable
```
[the remark this answers is not in the document]
Comment (hageboeck) [reply talk/tools/editors.tex:21]: I learned with eclipse, and I haven't seen anything that surpasses it. You can add CLion if you want.

[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/tools/editors.tex:21]: Wow, I did not know it was still this relevant. Ok, then we just keep it. CLion is already part of the list. I just thought to maybe replace Eclipse by it. Anyway, no change necessary then.

File: talk/tools/compiling.tex (hunk 1 above)
Comment (bernhardmgruber) [reply talk/tools/compiling.tex]: I mention `-O3` and `-Og` now.

**Review by hageboeck — APPROVED** [review]
For me you can merge.

**Pull request merged**